keiko-sql/src/main/kotlin/com/netflix/spinnaker/q/sql/SqlQueue.kt

```diff
@@ -281,7 +281,8 @@ class SqlQueue(
       return
     }
 
-    candidates.shuffle()
```

Comment:
SO MOSTLY concerned b/c if I'm reading the comments ABOVE correct:
```
 To minimize lock contention, this is a non-locking read. The id's returned may be
 locked or removed by another instance before we can acquire them. We read more id's
 than [maxMessages] and shuffle them to decrease the likelihood that multiple instances
 polling concurrently are all competing for the oldest ready messages when many more
 than [maxMessages] are read.
```
The shuffle shouldn't have mattered.  Except... it turns out it DOES matter b/c of lock behavior.  Wondering if we can update the comments or explanations on this... OR refactor this code to be less... confusing on how it operates in combination with below which does seem to use a lock mechanism...